### PR TITLE
Authorized Cards

### PR DESCRIPTION
**Pull request by techno-sam**
Allows creating sub-accounts with separate trust lists and specific spending limits to enable safer automated interaction with a bank account

### CODE REVIEW

File: common/src/main/java/dev/ithundxr/createnumismatics/base/item/SingleCallbackContainer.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package dev.ithundxr.createnumismatics.base.item;
+
+import net.minecraft.world.Container;
+import net.minecraft.world.ContainerListener;
+
+public class SingleCallbackContainer extends DelegatingContainer{
```
Comment (IThundxr) [inline common/src/main/java/dev/ithundxr/createnumismatics/base/item/SingleCallbackContainer.java]:
```suggestion
public class SingleCallbackContainer extends DelegatingContainer {
```

File: common/src/main/java/dev/ithundxr/createnumismatics/content/backend/trust_list/TrustListMenu.java [hunk 2]
```diff
@@ -163,7 +163,7 @@ protected boolean moveItemStackTo(ItemStack stack, int startIndex, int endIndex,
         if (reverseDirection) {
             i = endIndex - 1;
         }
-        if (stack.isStackable() && startIndex >= CARD_SLOTS) {
+        if (stack.isStackable() && startIndex >= CARD_SLOTS) { // CHANGED LINE
```
Comment (IThundxr) [inline common/src/main/java/dev/ithundxr/createnumismatics/content/backend/trust_list/TrustListMenu.java:166]: What's up with this?

File: common/src/main/java/dev/ithundxr/createnumismatics/util/Utils.java [hunk 3]
```diff
@@ -45,4 +48,12 @@ public static boolean testClientPlayerOrElse(Predicate<Player> predicate, boolea
     public static void openScreen(ServerPlayer player, MenuProvider factory, Consumer<FriendlyByteBuf> extraDataWriter) {
         throw new AssertionError();
     }
+
+    public static <R> void runOnceRegistered(ResourceKey<? extends Registry<R>> registryType, Runnable callback) {
```
Comment (IThundxr) [inline common/src/main/java/dev/ithundxr/createnumismatics/util/Utils.java:52]: This should probably use fabric's registry entry added event on fabric and fallback to registrate on forge

**Review by IThundxr — CHANGES_REQUESTED** [review]
Looks good, just a few changes

File: common/src/main/java/dev/ithundxr/createnumismatics/content/backend/trust_list/TrustListMenu.java (hunk 2 above)
Comment (techno-sam) [reply common/src/main/java/dev/ithundxr/createnumismatics/content/backend/trust_list/TrustListMenu.java:166]: I was just adding a marker on the only line that is actually changed from the vanilla impl

Comment (IThundxr) [reply common/src/main/java/dev/ithundxr/createnumismatics/content/backend/trust_list/TrustListMenu.java:166]: ah alright

File: common/src/main/java/dev/ithundxr/createnumismatics/util/Utils.java (hunk 3 above)
Comment (techno-sam) [reply common/src/main/java/dev/ithundxr/createnumismatics/util/Utils.java:52]: What's wrong with just using registrate on both sides? That way there's less of a chance of random buggyness - I'm not sure how I'd detect a registry already being registered w/ the fabric solution you propose

Comment (IThundxr) [reply common/src/main/java/dev/ithundxr/createnumismatics/util/Utils.java:52]: Fine for now i guess

**Review by IThundxr — APPROVED** [review]

**Review by IThundxr — APPROVED** [review]

**Pull request merged**